File: app/src/main/java/Bigbigdw/JoaraDW/Fragment_Finish/Fragment_Finish.java
package Bigbigdw.JoaraDW.Fragment_Finish;

import android.content.Context;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;
import androidx.viewpager.widget.ViewPager;

import com.google.android.material.tabs.TabLayout;

import Bigbigdw.JoaraDW.BookList.Finish_Tab;
import Bigbigdw.JoaraDW.R;
import Bigbigdw.JoaraDW.databinding.FragmentFinishBinding;

public class Fragment_Finish extends Fragment {

    private FragmentFinishBinding binding;

    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {

        binding = FragmentFinishBinding.inflate(inflater, container, false);
        View root = binding.getRoot();

        SectionsPagerAdapter sectionsPagerAdapter = new SectionsPagerAdapter(getContext(), getChildFragmentManager());
        ViewPager viewPager = binding.viewPager;
        viewPager.setAdapter(sectionsPagerAdapter);
        TabLayout Fragment_NewTab = binding.tabs;
        Fragment_NewTab.setupWithViewPager(viewPager);

        return root;
    }

    public static class SectionsPagerAdapter extends FragmentPagerAdapter {

        private final int[] TAB_TITLES = new int[]{R.string.Finish_Tab1, R.string.Finish_Tab2, R.string.Finish_Tab3, R.string.Finish_Tab4};
        private final Context mContext;

        public SectionsPagerAdapter(Context context, FragmentManager fm) {
            super(fm);
            mContext = context;
        }

        @Override
        public Fragment getItem(int position) {
            return Finish_Tab.newInstance(position + 1);
        }

        @Nullable
        @Override
        public CharSequence getPageTitle(int position) {
            return mContext.getResources().getString(TAB_TITLES[position]);
        }

        @Override
        public int getCount() {
            return TAB_TITLES.length;
        }
    }
}

File: app/src/main/java/bigbigdw/joaradw/fragment_finish/FragmentFinish.kt
package bigbigdw.joaradw.fragment_finish

import android.content.Context
import android.view.LayoutInflater
import android.view.ViewGroup
import android.os.Bundle
import android.view.View
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import bigbigdw.joaradw.R
import bigbigdw.joaradw.databinding.FragmentFinishBinding

class FragmentFinish : Fragment() {
    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        val fragmentFinishBinding = FragmentFinishBinding.inflate(inflater, container, false)
        val root: View = fragmentFinishBinding.root
        val sectionsPagerAdapter = SectionsPagerAdapter(
            context, childFragmentManager
        )
        val viewPager = fragmentFinishBinding.viewPager
        viewPager.adapter = sectionsPagerAdapter
        val fragmentNewTab = fragmentFinishBinding.tabs
        fragmentNewTab.setupWithViewPager(viewPager)
        return root
    }

    class SectionsPagerAdapter(private val mContext: Context?, fm: FragmentManager?) :
        FragmentPagerAdapter(
            fm!!
        ) {
        private val tabTitles = intArrayOf(
            R.string.Finish_Tab1,
            R.string.Finish_Tab2,
            R.string.Finish_Tab3,
            R.string.Finish_Tab4
        )

        override fun getItem(position: Int): Fragment {
            return FinishTab.newInstance(position + 1)
        }

        override fun getPageTitle(position: Int): CharSequence? {
            return mContext!!.resources.getString(tabTitles[position])
        }

        override fun getCount(): Int {
            return tabTitles.size
        }
    }
}